### PR TITLE
MCPAR July Release Updates (w/ Feature Flag)

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

This PR covers some of the requested changes for the July 25th MCPAR release, most of which are content updates (including adding a few new questions). Note that these updates are now behind the `julyMcparRelease` feature flag. 

* For [CMDCT-3717](https://jiraent.cms.gov/browse/CMDCT-3717), question D1.X.9 is being removed and replaced with three new questions: D1.X.9a, D1.X.9b, D1.X.9c
* For [CMDCT-3718](https://jiraent.cms.gov/browse/CMDCT-3718), question D1.X.6 is being expanded to include an additional radio button option
* For [CMDCT-3719](https://jiraent.cms.gov/browse/CMDCT-3719), three new questions are added to Topic IV. Appeals: D1.IV.1a, D1.IV.1b, D1.IV.1c

Additionally, I've moved all of the ILOS updates behind this feature flag because this will all be released together, so the `ilosEnabled` feature flag is going to be deprecated after this code is deployed into production.

### Related ticket(s)

CMDCT-3717
CMDCT-3718
CMDCT-3719

---
### How to test

The feature flag is turned on for `dev` and local environments, so you should see the latest version, including ILOS and the questions mentioned above. If you want to double-check for the opposite, just set [this variable](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/4321593a78101b898f6242372f59a3a8ae1d2fc8/services/ui-src/src/components/modals/AddEditReportModal.tsx#L44) to `false` ✨ 

 - Log in as a state user
 - Create a new MCPAR report
 - Add a new plan

Verify that the new questions are seen and match the Figma for each ticket (there's three separate tickets but the changes are pretty small in each one). Here's some screenshots to make reviews a bit smoother:

Route: `/mcpar/plan-level-indicators/program-integrity`

Screenshot 2024-06-26 at 2 58 40 PM
Screenshot 2024-06-26 at 2 58 32 PM

Route: `/mcpar/plan-level-indicators/appeals-state-fair-hearings-and-grievances/appeals-overview`

Screenshot 2024-06-26 at 3 00 10 PM

### Notes

`mcpar.json` has been updated.

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by gmrabian — DISMISSED** [review]

File: services/app-api/forms/mcpar.json [hunk 1]
```diff
@@ -3370,9 +3403,30 @@
                         }
                       ]
                     },
+                    {
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/app-api/forms/mcpar.json:3406]: Updated!

File: services/app-api/utils/formTemplates/formTemplates.ts [hunk 2]
```diff
@@ -318,17 +322,47 @@ const makePCCMTemplateModifications = (reportTemplate: ReportJson) => {
...
+        id: "plan_overpaymentRecoveryReportDescription",
+        type: "textarea",
+        validation: "text",
+        props: {
+          label: "D1.X.9 Plan overpayment reporting to the state",
+          hint: "Describe the plan’s latest annual overpayment recovery report submitted to the state as required under 42 CFR 438.608(d)(3).</br>Include, at minimum, the following information:<ul><li>The date of the report (rating period or calendar year).</li><li>The dollar amount of overpayments recovered.</li><li>The ratio of the dollar amount of overpayments recovered as a percent of premium revenue as defined in MLR reporting under 42 CFR 438.8(f)(2).</li></ul>",
+        },
+      };
+      route.children[6].drawerForm.fields.splice(7, 0, questionD1X9);
+    }
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/app-api/utils/formTemplates/formTemplates.ts:365]: great catch, I'll check that out. thank you!

**Comment by karla-vm** [comment]
> Might be a nit, but I'm seeing the spec text alignment differs from that of the PR
> 
> Spec: Screenshot 2024-06-27 at 11 24 17 AM
> 
> PR: Screenshot 2024-06-27 at 11 17 26 AM

I think that's expected since the Figma and the app tend to render a bit differently, but I'll ask Kim to make sure. thank you!

File: services/app-api/utils/formTemplates/formTemplates.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/app-api/utils/formTemplates/formTemplates.ts:369]: addressed this in the latest commit 💯

**Review by britt-mo — APPROVED** [review]

**Review by bangbay-bluetiger — APPROVED** [review]

**Review by jessabean — APPROVED** [review]

**Comment by karla-vm** [comment]
> I think they way you have it is correct but I want to double check.. The figma has slightly different wording for some of the D1.X.7 options:
> 
> Figma: Screenshot 2024-06-28 at 11 33 12 AM
> 
> Deployed Instance: Screenshot 2024-06-28 at 11 34 01 AM
> 
> Same issue with the D1.IV.1a question:
> 
> Figma: Screenshot 2024-06-28 at 11 38 17 AM
> 
> Deployed Instance: Screenshot 2024-06-28 at 11 37 43 AM

I just checked with Kim and that Figma you're referring to is actually a mock-up that will be shown to the BOs on a meeting this upcoming Monday, but it is not approved yet. So, for now we're using the Figma mock-up to the left of that.

**Pull request merged**